Modify isBbox regex to allow whitespace

trevorgerhardt opened the pull request:
This change allows optional whitespace at the beginning, end, and around numbers, ensuring that valid bbox strings are matched even if pasted with spaces.

Background: I pasted in a bbox string with whitespace which failed to match the original regex. The subsequent `JSON.parse` method failed and logged the issue to the console (maybe a separate issue to address?).

trevorgerhardt committed:
Modify isBbox regex to allow whitespace

This change allows optional whitespace around numbers and commas, ensuring that valid bbox strings are matched even if pasted with spaces.

Background: I pasted in a bbox string with whitespace which failed to match the original regex. The subsequent `JSON.parse` method failed and logged the issue to the console (maybe a separate issue to address?).

bdon commented:
Could we show a better error message via popup, etc to hint what a well-formed bbox is?

trevorgerhardt committed:
Add error handling to `loadTextArea`

Wrap `loadTextArea` in a `try {...} catch (e) {...}` to handle parsing errors for bboxes and GeoJSON. Show the textarea as invalid with a red border when there is a parsing error. Display the error text below the <textarea>.

trevorgerhardt commented:
Certainly. The main issue is that if the regex fails, the method then attempts to parse it as GeoJSON and when that fails the error is thrown. So it's not obvious that it was a bad bbox or bad GeoJSON. But we can make a guess.

In https://github.com/SliceOSM/sliceosm-frontend/pull/18/commits/b333b96c8814c30b8b8b5e4ac49d009b1ef73598 I have implemented an error handler around `loadTextArea` that catches the errors, logs them to console, and then makes a guess for which type of error it might be:
- `SyntaxError` from JSON.parse, show bbox helper text
- Otherwise show GeoJSON advice

Example `SyntaxError` state:
<img width="334" alt="Screenshot 2025-04-01 at 09 31 24" src="https://github.com/user-attachments/assets/f7fe4934-f537-41b7-bff7-fd11213c15ee" />


Additional options requiring more changes but might make for a better UX or easier, more robust error handling:
1. Split up the <textarea> into separate "Paste bbox" and "Paste GeoJSON" textareas with separate load buttons and handling.
2. OR separate just the button into two "Load bbox" and "Load GeoJSON" buttons that use the same textarea but separate parsing and error handling.
4. Add https://github.com/placemark/check-geojson in order to do a more thorough error checking when parsing GeoJSON.

bdon commented:
Thanks, it's live now on https://slice.openstreetmap.us

additional PRs to make the error handling better are welcome
